src/utils/extensionUtils.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Event, EventEmitter, extensions } from "vscode";
+
+export class ExtensionUtils {
+    private extensionName: string;
+    private readonly extensionInstalledEmitter: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    public readonly onExtensionInstalled: Event<boolean> = this.extensionInstalledEmitter.event;
+
+    constructor(extensionName: string) {
+        this.extensionName = extensionName;
+        this.subscribeToExtensionChange();
+    }
+
+    private subscribeToExtensionChange(): void {
+        if (this.isExtensionInstalled(this.extensionName)) {
+            this.extensionInstalledEmitter.fire(true);
+        }
+
+        extensions.onDidChange(() => {

```

Comment:
This returns a disposable object (with which to unsubscribe from the event), which should be disposed of when we no longer care about listening.